### PR TITLE
Isu 10 error when respondent tries to login without verifying activating account

### PR DESCRIPTION
**Pull request by nherriot**
This puts in the logic to have the OAuth2 server reject a request for a token if the account is not verified. There are also a number of new test cases added to test boundary conditions on accounts not being verified to not verified as well combinations of accounts being locked.

### CODE REVIEW

File: oauth2server/apps/tokens/tests/test_user_credentials.py [hunk 1]
```diff
@@ -87,6 +87,102 @@ def test_corrupt_client_credentials(self):
...
+            HTTP_AUTHORIZATION='Basic:{}'.format(
+                base64.encodestring('testclient:testpassword')),
+        )
+
+        self.assertEqual(OAuthAccessToken.objects.count(), 0)
+        self.assertEqual(OAuthRefreshToken.objects.count(), 0)
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(response.data['detail'], u'User account not verified')
+        #print "****** response value is: {} *******".format(response.data)
```
Comment (liamtoozer) [inline oauth2server/apps/tokens/tests/test_user_credentials.py]: Do the print statements in here need to be removed?

File: oauth2server/apps/tokens/decorators.py [hunk 2]
```diff
@@ -196,6 +197,10 @@ def _validate_grant_type(request):
                 stdlogger.warning( "Raised InvalidUserCredentialsException")
                 raise InvalidUserCredentialsException()
 
+            if not user.account_is_verified:
+                stdlogger.warning("Raised UserAccountNotVerified")
+                raise UserAccountNotVerified
```
Comment (liamtoozer) [inline oauth2server/apps/tokens/decorators.py]: Should `raise UserAccountNotVerified` have parentheses in the call to be consistent with other raised exceptions here? Still works fine however.

**Review by liamtoozer — CHANGES_REQUESTED** [review]
Looks good to me. I'm getting `{"detail":"User account not verified"}` in my 401 response as expected. 

Just a couple of tiny suggestions for changes.

File: oauth2server/apps/tokens/tests/test_user_credentials.py (hunk 1 above)
Comment (nherriot) [reply oauth2server/apps/tokens/tests/test_user_credentials.py]: done.

File: oauth2server/apps/tokens/decorators.py (hunk 2 above)
Comment (nherriot) [reply oauth2server/apps/tokens/decorators.py]: Good catch! still works but don't know why... fixed now, and also removed some white space.
